app/javascript/components/request-workflow-status/data.js

```diff
@@ -0,0 +1,216 @@
+import moment from 'moment';
+
+/** Types of workflow state status */
+export const workflowStateTypes = {
+  success: { text: 'success', tagType: 'green' },
+  error: { text: 'error', tagType: 'red' },
+  failed: { text: 'failed', tagType: 'gray' },
+};
+
+/** Function to get the header data of workflow states table. */
+const headerData = () => ([
+  {
+    key: 'name',
+    header: 'Name',
+  },
+  {
+    key: 'enteredTime',
+    header: 'Entered Time',
+  },
+  {
+    key: 'finishedTime',
+    header: 'Finished Time',
+  },
+  {
+    key: 'duration',
+    header: 'Duration',
+  },
+]);
+
+const convertDate = (date) => `${moment(date).format('MM/DD/YYYY')} ${moment(date).format('h:mm:ss A')}`;
+
+const duration = (StartTime, FinishedTime) => {
+  const start = moment(StartTime);
+  const end = moment(FinishedTime);
+  const diff = end.diff(start);
+  return `${moment.utc(diff).format('.SSS')}s`;
+};
+
+/** Function to get the row data of workflow states table. */
+const rowData = ({ States }) => States.map((item) => ({
+  id: item.Guid,
+  name: item.Name,
+  enteredTime: convertDate(item.EnteredTime),
+  finishedTime: convertDate(item.FinishedTime),
+  duration: duration(item.StartTime, item.FinishedTime),
+}));
+
+/** Dummy data received from api response. */
```

Comment:
Marking this PR as WIP until we remove all the dummy data and it's actually using proper API

Reply:
Hey @Fryguy, from where can we extract the `configuration_script_id` that can be passed to the api. 
the `recordId` passed into this component is the MiqRequest's id.